### PR TITLE
Add unit tests and perform dependency updates for the metrics project

### PR DESCRIPTION
**Pull request by stephenfuqua**
## Unit Testing Summary

Adds a unit test suite for the `edfi_tech_metrics` Python package, covering pure/logic functions that do not require mocking.

### Changes

- **`metrics/tests/test_settings.py`** - 8 tests for `Configuration` logging behavior and `load_configuration()` argument parsing
- **`metrics/tests/test_ossf_score.py`** - 7 tests for `compare_with_previous_scores()` (new/removed/changed repos, score delta, edge cases)
- **`metrics/tests/test_improvements.py`** - 5 tests for `calculate_improvements()` using `tmp_path` fixture

### Refactors / Bug Fixes

- `ossf_score.py`: added `data_dir` parameter to `compare_with_previous_scores()` so tests can pass a temp directory
- `settings.py`: fixed `parsed.jira_user_name` to `parsed.user_name` (configargparse derives attr name from `--user-name`, not the env var name)
- `pyproject.toml`: added `pytest = "^8.0"` to dev dependencies

## Dependency Updates

| Package | From | To | Type | Notes |
|---|---|---|---|---|
| plotnine | 0.15.3 | 0.15.4 | patch | direct dep |
| types-requests | 2.33.0.20260408 | 2.33.0.20260513 | patch | dev dep |
| pandas | 2.3.3 | 3.0.3 | **major** | pytz removed as transitive dep |
| mypy | 1.20.2 | 2.1.0 | **major** | new dep: ast-serialize |
| pytest | 8.4.2 | 9.0.3 | **major** | |
| numpy | 2.4.0 | 2.4.4 | minor | transitive |
| scipy | 1.16.3 | 1.17.1 | minor | transitive |
| matplotlib | 3.10.8 | 3.10.9 | minor | transitive |
| requests | 2.33.0 | 2.34.2 | minor | transitive |
| ipython | 9.8.0 | 9.13.0 | minor | transitive |
| certifi | 2025.11.12 | 2026.4.22 | minor | transitive |
| anyio | 4.12.0 | 4.13.0 | minor | transitive |
| attrs | 25.4.0 | 26.1.0 | minor | transitive |
| packaging | 25.0 | 26.2 | minor | transitive |
| tzdata | 2025.3 | 2026.2 | minor | transitive |
| pycparser | 2.23 | 3.0 | **major** | transitive only (no direct usage) |
| *(+38 other transitive minor/patch updates)* | | | minor/patch | all within existing constraints |

## Test Plan

- [x] 20/20 tests passing (`pytest metrics/tests/`)
- [x] No new lint/type errors introduced

### CODE REVIEW

**Comment by stephenfuqua** [comment]
Testing: I've run the code locally to confirm that everything is working correctly still.

File: metrics/tests/test_settings.py [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+
+
+def test_load_configuration_returns_correct_fields():
+    args = ["-u", "testuser", "-t", "testtoken", "-b", "http://jira.example.com"]
+    conf = load_configuration(args)
+    assert conf.jira_user_name == "testuser"
+    assert conf.jira_token == "testtoken"
+    assert conf.jira_base_url == "http://jira.example.com"
+    assert conf.log_level == "INFO"
+    assert conf.page_size == 100
```
[the remark this answers is not in the document]
Comment (stephenfuqua) [reply metrics/tests/test_settings.py]: I'm not worried about this right now. Going to ignore the suggestion.

File: .github/workflows/test-metrics.yml [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+          cache: "poetry"
+
+      - name: Install dependencies
+        run: poetry env use "3.12" && poetry install
+
+      - name: Lint
+        run: poetry run flake8
+
+      - name: Type Check
+        if: always()
```
[the remark this answers is not in the document]
Comment (stephenfuqua) [reply .github/workflows/test-metrics.yml:48]: Interestingly, the comment above does highlight a real problem... but mypy isn't detecting it. Copilot is fixing that locally.

[the remark this answers is not in the document]
Comment (stephenfuqua) [reply .github/workflows/test-metrics.yml:48]: Copilot (locally) agrees with me that this is a bad suggestion above, because it would cause this failure to be treated as a warning.

>  For lint/type-check/test steps specifically: using continue-on-error: true would be counterproductive — you want  those to fail the build. So if a code review comment is suggesting adding continue-on-error: true to these steps, I'd push back on it.

File: metrics/edfi_tech_metrics/ossf_score.py [hunk 3]
```diff
@@ -48,12 +48,11 @@ def fetch_repositories(organization):
         return []
 
 
-def compare_with_previous_scores(current_scores):
+def compare_with_previous_scores(current_scores, data_dir: str = "./data/ossf-scores/"):
```
[the remark this answers is not in the document]
Comment (stephenfuqua) [reply metrics/edfi_tech_metrics/ossf_score.py]: fixed

File: metrics/tests/test_ossf_score.py [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+
+import math
+
+import pandas as pd
+import pytest
+
+from edfi_tech_metrics.ossf_score import compare_with_previous_scores
+
+
+def test_no_previous_files_prints_message(tmp_path, capsys):
```
[the remark this answers is not in the document]
Comment (stephenfuqua) [reply metrics/tests/test_ossf_score.py]: fixed

File: metrics/tests/test_improvements.py [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+from edfi_tech_metrics.improvements import calculate_improvements
+
+
+def _write_stats_csv(path: Path, date: str, rows: list) -> None:
+    for row in rows:
+        row["date"] = date
+    pd.DataFrame(rows).to_csv(path, index=False)
+
+
+def test_improvement_positive_delta(tmp_path):
```
[the remark this answers is not in the document]
Comment (stephenfuqua) [reply metrics/tests/test_improvements.py]: fixed

File: metrics/tests/test_settings.py [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+# Licensed to the Ed-Fi Alliance under one or more agreements.
+# The Ed-Fi Alliance licenses this file to you under the Apache License, Version 2.0.
+# See the LICENSE and NOTICES files in the project root for more information.
+
+import pytest
+
+from edfi_tech_metrics.settings import Configuration, load_configuration
+
+
+def test_info_prints_at_info_level(capsys):
```
[the remark this answers is not in the document]
Comment (stephenfuqua) [reply metrics/tests/test_settings.py]: fixed

**Review by stevenarnold-ed-fi — APPROVED** [review]
